client/components/data/query-sites/index.jsx

```diff
@@ -51,15 +95,31 @@ QuerySites.propTypes = {
 
 QuerySites.defaultProps = {
 	allSites: false,
+	primaryAndRecent: false,
 	requestSites: () => {},
 	requestSite: () => {},
 };
 
 export default connect(
 	( state, { siteId } ) => {
+		const primarySiteId = getPrimarySiteId( state );
+		const recentSiteIds = getPreference( state, 'recentSites' );
+		const recentSiteIdsWithoutPrimary = without(
+			getPreference( state, 'recentSites' ),
+			primarySiteId
+		);
+
 		return {
+			primarySiteId,
+			recentSiteIds,
+			hasAllSitesList: hasAllSitesList( state ),
 			requestingSites: isRequestingSites( state ),
 			requestingSite: isRequestingSite( state, siteId ),
+			isRequestingPrimarySite: isRequestingSite( state, primarySiteId ),
+			isRequestingRecentSites: some(
+				recentSiteIdsWithoutPrimary,
+				isRequestingSiteFactory( state )
```

Comment:
I would just add an eslint comment to disable the `wpcalypso/redux-no-bound-selectors` here and inline the arrow functions. The eslint rule is buggy and shoudn't really complain here.

Reply:
Ok.